### DISCUSSION TITLE
Convert images but keep the original file name

### DISCUSSION THREAD

**Discussion by gnoatx** [ImageMagick/ImageMagick#6494] [Help]
### ImageMagick version

7.1.1-13

### Operating system, version and so on

Windows 11

### Description

Is there a simpler way to convert images to another format while keeping the filename from the source (except for the extension)?
Like my brain instantly told me to do
`magick *.jpg *.png`
because most basic imagemagick commands are super simple and intuitive, but that doesn't work. After a bunch of googling i found a way that's
`magick *.jpg -set filename:base "%[basename]" "%[filename:base].png"`
but that feels a little silly to me that such a simple concept would have such a complicated solution

**Comment by fmw42** [comment]
That is the correct solution.  magick (or convert) does not allow wild card output.

Alternately you can use mogrify to process a whole directory of images (or just one) and change suffix without changing the base name.

**Reply by fmw42** [reply]
See https://imagemagick.org/Usage/basics/#mogrify

**Reply by gnoatx** [reply]
I see. Is there a reason why this is the way it is?

**Reply by fmw42** [reply]
Convert/Magick is to complex to allow wildcard outputs.

I suggest you use mogrify.
